prompt2model/utils/api_tools.py

```diff
@@ -219,22 +220,38 @@ async def _throttled_completion_acreate(
         return responses
 
 
-def handle_api_error(e) -> None:
+def handle_api_error(e, backoff_duration=1) -> None:
     """Handle OpenAI errors or related errors that the API may raise.
 
+    Sleeps incase error is some type of timeout, else throws error.
+
     Args:
         e: The error to handle. This could be an OpenAI error or a related
            non-fatal error, such as JSONDecodeError or AssertionError.
+        backoff_duration: The duration to wait before retrying the API call.
+
+    Raises:
+        e: If the error is not an instance of APIError, Timeout, or RateLimitError.
+
+    Returns:
+        None
     """
     logging.error(e)
-    if not isinstance(e, API_ERRORS):
+
+    if not isinstance(e, openai.error.OpenAIError):
         raise e
+
     if isinstance(
-        e,
-        (openai.error.APIError, openai.error.Timeout, openai.error.RateLimitError),
+        e, (openai.error.APIError, openai.error.Timeout, openai.error.RateLimitError)
     ):
-        # For these errors, OpenAI recommends waiting before retrying.
-        time.sleep(1)
+
+        match = re.search(r"Please retry after (\d+) seconds", str(e))
+        # If openai mentions how long to sleep use that, else do exponential backoff
```

Comment:
Make this a complete sentence.
```suggestion
        # If OpenAI mentions how long to sleep, use that. Otherwise, do
        # exponential backoff.
```